cmd/kata-manager/kata-manager.sh

```diff
@@ -0,0 +1,258 @@
+}
+
+disable_initrd()
+{
+	local file="$1"
+
+	sudo sed -i 's/^\(initrd =.*\)/# \1/g' "$config_file"
+}
+
+enable_image()
+{
+	local file="$1"
+
+	sudo sed -i "s/^#*image *=/image = \"$file\"" "$config_file"
+}
+
+enable_initrd()
+{
+	local file="$1"
+
+	sudo sed -i "s/^#*initrd *=/initrd = \"$file\"" "$config_file"
+}
+
+cmd_enable_full_debug()
+{
+	info "enabling debug"
+
+	sudo sed -i -e 's/^# *\(enable_debug\).*=.*$/\1 = true/g' "$config_file"
+	sudo sed -i -e 's/^kernel_params = "\(.*\)"/kernel_params = "\1 ${agent_debug}"/g' "$config_file"
+}
+
+cmd_disable_all_debug()
+{
+	info "disabling debug"
+
+	sudo sed -i -e 's/^\(enable_debug.*=.*$\)/# \1/g' "$config_file"
+	sudo sed -i -e 's/^\(kernel_params = ".*\)${agent_debug}\(.*"\)/\1 \2/g' "$config_file"
+}
+
+cmd_configure_image()
+{
+	local file="$1"
+
+	check_file "$file"
+
+	info "installing '$file' as image"
+
+	enable_image "$file"
+	disable_initrd "$file"
+}
+
+cmd_configure_initrd()
+{
+	local file="$1"
+
+	check_file "$file"
+
+	info "installing '$file' as initrd"
+
+	enable_initrd "$file"
+	disable_image "$file"
+}
+
+# Install the packaged version of Kata by executing the commands specified in the
+# installation guide document.
+cmd_install_packages()
+{
+	[ -z "${GOPATH}" ] && die "need GOPATH"
+
+	source /etc/os-release
+
+	local doc_repo_url="https://${doc_repo}"
+
+	local repo_dir="${GOPATH}/src/${doc_repo}"
+
+	info "installing packages"
+
+	[ -d "${repo_dir}" ] || (cd "$(dirname ${repo_dir})" && git clone "$doc_repo_url")
+
+	local file="${distro}-installation-guide.md"
+
+	local doc="${GOPATH}/src/${doc_repo}/install/${file}"
+	[ -e "$doc" ] || die "no install document for distro $distro"
+
+	local -r doc_script="kata-doc-to-script.sh"
+
+	local tool="${GOPATH}/src/${test_repo}/.ci/${doc_script}"
+	[ -e "${tool}" ] || die "cannot find script $doc_script"
+
+	local install_script=$(mktemp)
+
+	# create the script
+	"${tool}" "${doc}" "${install_script}"
+
+	# run the installation
+	bash "${install_script}"
+
+	# clean up
+	rm -f "${install_script}"
+}
+
+cmd_remove_packages()
+{
+	local packages_regex="^(kata|qemu-lite)-"
+	local packages
+
+	info "removing packages"
+
+	if [ "$distro" = "centos" ]
+	then
+		packages=$(rpm -qa|egrep "${packages_regex}")
+		[ -z "$packages" ] && { echo "packages not installed" && exit 0; }
+
+		sudo yum -y remove $packages
+	elif [ "$distro" = "fedora" ]
+	then
+		packages=$(rpm -qa|egrep "${packages_regex}")
+		[ -z "$packages" ] && { echo "packages not installed" && exit 0; }
+
+		sudo dnf -y remove $packages
+	elif [ "$distro" = "ubuntu" ]
+	then
+		packages=$(dpkg-query -W -f='${Package}\n'|egrep "${packages_regex}")
+		[ -z "$packages" ] && { echo "packages not installed" && exit 0; }
+
+		sudo apt-get -y remove $packages
+	else
+		die "invalid distro: '$distro'"
+	fi
+}
+
+setup()
+{
+	source /etc/os-release
+
+	# lower-case distribution name
+	distro=$(echo "$NAME"|tr '[A-Z]' '[a-z]')
```

Comment:
I think we should better use `$ID` instead of `$NAME`.
In CentOS:
`NAME="CentOS Linux"`

Reply:
Another good catch - fixed.